apps/s_time.c

```diff
@@ -240,16 +224,10 @@ int s_time_main(int argc, char **argv)
                                    www_path);
             if (buf_len <= 0 || SSL_write(scon, buf, buf_len) <= 0)
                 goto end;
-            while ((i = SSL_read(scon, buf, sizeof(buf))) > 0 ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_READ ||
-                        SSL_get_error(scon, i) == SSL_ERROR_WANT_WRITE)
```

Comment:
Hmm, I see.

My gut feeling this ERROR_WANT_READ did not happen without
the other side doing an explicit SSL_renegotiate, which is rarely the case,
and it is likely to break some more code out there.

I added SSL_CTX_set_mode(ctx, SSL_MODE_AUTO_RETRY)
but I got a new problem, the renegotiation does not work with TLS 1.3.

Could you have a look:
I build with ./config -d no-pic enable-tls1_3
./openssl s_server
sing default temp DH parameters
ACCEPT
ERROR
shutting down SSL
CONNECTION CLOSED
ACCEPT

./openssl s_time -reuse
No CIPHER specified


Now timing with session id reuse.
starting
ERROR